gradio-app/gradio
# Add `show_api` parameter to events, and fix `gr.load()`. Also makes some minor improvements to the "view API" page when running on Spaces

This PR does several things

1. It adds a `show_api` parameter to each event trigger. This `show_api` parameter controls whether or not the API route is visible in the API docs. Specifically, we set to `False` for internal routes in `gr.Interface()` and `gr.ChatInterface()` instead of setting `api_name=False`, which was causing issues using `gr.load()` with `gr.Interface()` and `gr.ChatInterface()`.

Fixes: #6744
Fixes: #6476 

2. Disallows the `api_name=""`, which was causing issues. If `api_name=""` is passed in, it is treated the same as `api_name=None`, and the function name is used instead. 

Fixes: #4894 

3. Also makes some minor improvements to the View API docs pages. For example, "Named endpoints" is replaced with "endpoints", endpoints are counted correctly, and the whitespace is fixed. 

------

**Testing instructions:**

I've installed this PR in: https://huggingface.co/spaces/gradio/test-gr-load

You can confirm this works by `gr.load()`-in the Space as follows and making sure the loaded Space works fully:

```py
import gradio as gr

io = gr.load("spaces/gradio/test-gr-load")
io.launch()
```
